src/lib/ingest.js

```diff
@@ -1,51 +1,148 @@
-import { Readable } from 'readable-stream'
-import pump from 'pump'
+import { getItemCreated } from './database.js'
+import { dbClient, createIndex } from './databaseClient.js'
 import logger from './logger.js'
 
-export async function ingestItem(item, stream) {
-  const readable = new Readable({ objectMode: true })
-  const { toDB, dbStream } = await stream()
-  const promise = new Promise((resolve, reject) => {
-    pump(
-      readable,
-      toDB,
-      dbStream,
-      (error) => {
-        if (error) {
-          logger.error('Error ingesting', error)
-          reject(error)
-        } else {
-          logger.info(`Ingested item ${item.id}`)
-          resolve(true)
-        }
-      }
-    )
+const COLLECTIONS_INDEX = process.env['COLLECTIONS_INDEX'] || 'collections'
+
+export async function convertIngestObjectToDbObject(
+  // eslint-disable-next-line max-len
+  /** @type {{ hasOwnProperty: (arg0: string) => any; collection: string; links: any[]; id: any; }} */ data
+) {
+  let index = ''
```

Comment:
Filed this issue https://github.com/stac-utils/stac-server/issues/486 against using this heuristic (I realize it was copied from databaseStream.js, so it was pre-existing). You're welcome to fix that if you want.